data_utils.py

```diff
@@ -63,118 +62,74 @@ def get_dataset_loader(config, dataset, shuffle=False, train=True):
-    else:
-        datasets = _load_raw_data(config, data_dir)
-        for split in datasets.keys():
-            datasets[split] = _preprocess_on_split(datasets[split])
-        with open(cache_path, 'wb') as fp:
-            pickle.dump(datasets, fp)
-
-    map_path = os.path.join(data_dir, f'vocab_label_map_{config["min_vocab_freq"]}.pkl')
-    map_path = config['vocab_label_map'] or map_path
-    if map_path and os.path.exists(map_path):
-        log.info(f'Load existing Vocab and classes from {map_path}.')
-        with open(map_path, 'rb') as fp:
-            text_dict, classes = pickle.load(fp)
-    else:
-        classes = set()
-        for dataset in datasets.values():
-            for d in tqdm.tqdm(dataset):
-                classes.update(d['label'])
-        classes = sorted(classes)
-
-        text_dict = build_text_dict(datasets['train'], config['min_vocab_freq'], config['vocab_file'])
-        with open(map_path, 'wb') as fp:
-            pickle.dump((text_dict, classes), fp)
-        log.info(f'Save {map_path}')
+def _load_raw_data(path):
+    log.info(f'Load data from {path}.')
+    data = pd.read_csv(path, sep='\t', names=['label', 'text'],
+                       converters={'label': lambda s: s.split(),
+                                   'text': tokenize})
+    data = data.reset_index().to_dict('records')
+    data = [d for d in data if len(d['label']) > 0]
+    return data
 
-    if 'dev' not in datasets:
-        dev_size = config['dev_size'] if config['dev_size'].is_integer() else int(config['dev_size'] * len(datasets['train']))
-        train_size = len(datasets['train']) - dev_size
-        datasets['train'], datasets['dev'] = torch.utils.data.random_split(datasets['train'], [train_size, dev_size], generator=torch.Generator().manual_seed(42))
 
-    for split in datasets.keys():
-        datasets[split] = TextDataset(datasets[split], text_dict, classes, config['max_seq_length'])
+@log.enter('load_dataset')
+def load_datasets(config):
+    datasets = {}
+    test_path = config.test_path or os.path.join(config.data_dir, 'test.txt')
+    if os.path.exists(test_path):
+        datasets['test'] = _load_raw_data(test_path)
+    if config.eval:
+        return datasets
+
+    train_path = config[f'train_path'] or os.path.join(config.data_dir, 'train.txt')
+    datasets['train'] = _load_raw_data(train_path)
+    val_path = config[f'val_path'] or os.path.join(config.data_dir, 'valid.txt')
+    if os.path.exists(val_path):
+        datasets['val'] = _load_raw_data(val_path)
+    else:
+        datasets['train'], datasets['val'] = train_test_split(
+            datasets['train'], test_size=config.val_size, random_state=42)
 
-    log.info(f"Finish loading dataset (train: {len(datasets['train'])} / test: {len(datasets['test'])} / dev: {len(datasets['dev'])})")
+    log.info(f"Finish loading dataset (train: {len(datasets['train'])} / test: {len(datasets['test'])} / val: {len(datasets['val'])})")
```

Comment:
What if the user does not specify a `test.txt` in the training mode?

Reply:
The program will do nothing after training.
Please check L138 in main.py.

Reply:
Yes, but in this `log.info` there will not have`test` key in the `datasets` which give a `KeyError`?

Reply:
Oops, you're right!